### PR TITLE
dts: riscv: Add sifive,plic-1.0.0 binding and fix riscv,ndev values

### PR DESCRIPTION
**Pull request by ulfalizer**
Add a new sifive,plic-1.0.0 binding that inherits from the riscv,plic0
binding. The new binding adds a required riscv,ndev property, which
gives the number of external interrupts supported.

Use the new binding for microsemi-miv.dtsi (with a value of 31 for
riscv,ndev, from http://www.actel.com/ipdocs/MiV_RV32IMAF_L1_AHB_HB.pdf)
and riscv32-fe310.dtsi (which already assigns riscv,ndev).

Also remove a spurious riscv,ndev assignment from
riscv32-litex-vexriscv.dtsi.

Also make edtlib and the old scripts/dts/ scripts replace '.' in
compatible strings with '_' when generating identifiers.

### CODE REVIEW

**Review by galak — CHANGES_REQUESTED** [review]
I think `riscv,ndev` should only be on the sifive,plic.  It should probably be removed from `dts/riscv32/riscv32-litex-vexriscv.dtsi`

**Review by nategraff-sifive — APPROVED** [review]
Approving the PLIC property; I can't comment on the VexRiscv interrupt controller.

**Comment by ulfalizer** [comment]
@galak 
Changed

**Review by galak — CHANGES_REQUESTED** [review]
I think I was going to say we should match the Linux bindings a introduce a `sifive,plic-1.0.0` binding that has `riscv,ndev` and inherits from `riscv,plic0.yaml`

**Comment by ulfalizer** [comment]
@galak 
Updated

**Comment by ulfalizer** [comment]
Got help from @PiotrZierhoffer and @mateusz-holenko re. the MiV board.

File: dts/riscv32/microsemi-miv.dtsi [hunk 1]
```diff
@@ -44,14 +44,15 @@
 
 		plic: interrupt-controller@40000000 {
 			#interrupt-cells = <1>;
-			compatible = "riscv,plic0";
+			compatible = "sifive,plic-1.0.0";
```
Comment (galak) [inline dts/riscv32/microsemi-miv.dtsi]: Just checking, its the case that microsemi-miv is using SiFive PLIC?

Comment (PiotrZierhoffer) [reply dts/riscv32/microsemi-miv.dtsi]: Yes, that's correct

**Review by galak — COMMENTED** [review]
You made need to update `soc/riscv32/riscv-privilege/miv/dts_fixup.h` and `drivers/interrupt_controller/plic.c`

**Review by galak — APPROVED** [review]

**Comment by ulfalizer** [comment]
> You made need to update `soc/riscv32/riscv-privilege/miv/dts_fixup.h` and `drivers/interrupt_controller/plic.c`

Done

**Review by carlescufi — APPROVED** [review]

**Comment by galak** [comment]
How'd you get 31 for riscv,ndev on microsemi-miv.dtsi?

**Comment by mateusz-holenko** [comment]
According to this doc: https://www.microsemi.com/document-portal/doc_download/1243781-mi-v-rv32imaf-l1-axi-handbook, page 10, Table 2 "MiV_RV32IMAF_L1_AHB Architecture" there are 31 external interrupts for MiV.

**Comment by galak** [comment]
> According to this doc: [microsemi.com/document-portal/doc_download/1243781-mi-v-rv32imaf-l1-axi-handbook](https://www.microsemi.com/document-portal/doc_download/1243781-mi-v-rv32imaf-l1-axi-handbook), page 10, Table 2 "MiV_RV32IMAF_L1_AHB Architecture" there are 31 external interrupts for MiV.

So is NUM_IRQS incorrect in soc/riscv/riscv-privilege/miv/Kconfig.defconfig.series ?

**Pull request merged**